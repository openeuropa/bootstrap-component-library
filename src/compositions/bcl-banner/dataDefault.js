const drupalAttribute = require("drupal-attribute");

module.exports = {
  title: "EU Budget for the future",
  description: "Innovation, economy, environment and geopolitics",
  link: {
    label: "Subscribe",
    path: "/example.html",
    icon: {
      name: "chevron-right",
      path: "boostrap-icons.svg",
    },
    attributes: new drupalAttribute().addClass(["btn", "btn-primary"]),
  },
  attributes: new drupalAttribute().addClass([
    "bg-light",
    "text-dark",
    "text-center",
  ]),
};
