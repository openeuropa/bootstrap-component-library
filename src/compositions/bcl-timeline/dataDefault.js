const drupalAttribute = require("drupal-attribute");

module.exports = {
  id: 1,
  toggle_collapsed: "Show %d more items",
  toggle_expanded: "Hide %d items",
  hide: {
    from: 5,
    to: -2,
  },
  icons_path: "boostrap-icons.svg",
  items: [
    {
      label: "13 September 2017",
      title: "Item title",
      content:
        '<a href="/example">President Juncker\'s State of the Union speech</a>',
    },
    {
      label: "28-29 September 2017",
      content: '<a href="/example">Informal Digital Summit, Tallinn</a>',
    },
    {
      label: "14 November 2017",
      content:
        "Strengthening European identity through education and culture: European Commission's contribution to the Leaders' meeting.",
    },
    {
      label: "17 November 2017",
      content: '<a href="/example">Social Summit in Gothenburg, Sweden</a>',
    },
    {
      label: "6 December 2017",
      content:
        '<a href="/example">Economic and Monetary Union package of proposals</a>',
    },
    {
      label: "14-15 December 2017",
      content:
        '<a href="/example">EU Leaders\' meeting on migration, Brussels</a>',
    },
    {
      label: "15 December 2017",
      content: '<a href="/example">Euro Summit</a>',
    },
    {
      label: "6 February 2018",
      content: '<a href="/example">EU-Western Balkans Strategy</a>',
    },
    {
      label: "14 February 2018",
      content:
        '<a href="/example">Multiannual Financial Framework</a> and <a href="/example">institutional issues</a> - enhancing efficiency at the helm of the European Union',
    },
    {
      label: "23 February 2018",
      content: '<a href="/example">Informal European Council</a>',
    },
    {
      label: "22-23 March 2018",
      content: '<a href="/example">European Council</a>',
    },
    {
      label: "2 May 2018",
      content:
        '<a href="/example">Commission proposal: Long-term budget post-2020</a>',
    },
    {
      label: "15 May 2018",
      content:
        '<a href="/example">European Commissionâ€™s contribution to the Informal Leaders\' meeting</a>, Sofia, Bulgaria (16 May 2018)',
    },
    {
      label: "17 May 2018",
      content:
        '<a href="/example">EU-Western Balkans Summit</a> (Sofia, Bulgaria)',
    },
    {
      label: "23 May 2018",
      content:
        'Progress by Member States in meeting the conditions for adopting the euro - <a href="/example">reports</a> by the European Commission and ECB',
    },
    {
      label: "23 May 2018",
      content:
        '<a href="/example">European Semester</a>: European Commission publishes economic policy guidance for Member States',
    },
    {
      label: "29 May - 14 June 2018",
      content:
        '<a href="/example">EU budget for the future</a>: European Commission\'s legislative proposals for spending programmes',
    },
    {
      label: "28-29 June 2018",
      content:
        'Commission contribution to the European Council: <a href="/example">Economic and Monetary Union, migration, EU budget for the future</a>',
    },
    {
      label: "10 July 2018",
      content:
        'Ways to enhance subsidiarity, proportionality and better regulation in the daily operations of the EU: <a href="/example">Report</a> by the Task Force chaired by Commission First Vice-President Timmermans',
    },
    {
      label: "12 September 2018",
      content:
        '<a href="/example">President Juncker\'s State of the Union speech</a>',
    },
    {
      label: "20 September 2018",
      content:
        '<a href="/example">Informal European Council/Meeting of EU27 (Salzburg, Austria)</a>',
    },
    {
      label: "18 October 2018",
      content: '<a href="/example">European Council meeting</a>, Brussels',
    },
    {
      label: "21 November 2018",
      content:
        '<a href="/example">European Semester</a>: setting the economic and social priorities',
    },
    {
      label: "28 November 2018",
      content:
        'Commission contribution to the European Council on <a href="/example">Capital Markets Union and Banking Union</a>',
    },
    {
      label: "04 December 2018",
      content:
        'Commission contribution to the European Council on <a href="/example">long-term EU budget</a> and <a href="/example">migration</a>',
    },
    {
      label: "05 December 2018",
      content:
        'Commission contribution to the European Council on the <a href="/example">international role of the euro</a> and the <a href="/example">Action Plan on disinformation</a>',
    },
    {
      label: "11 December 2018",
      content:
        'Commission contribution to the European Council on the <a href="/example">Citizens\' Dialogues</a>',
    },
    {
      label: "11 December 2018",
      content:
        'Commission contribution to the <a href="/example">Euro Summit</a>',
    },
    {
      label: "13-14 December 2018",
      title: "Item title",
      content: "European Council and Euro Summit",
    },
  ],
  attributes: new drupalAttribute(),
};
