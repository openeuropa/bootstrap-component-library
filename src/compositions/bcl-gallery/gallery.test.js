import { renderTwigFileAsNode } from "@openeuropa/bcl-test-utils";
import dataDefault from "@openeuropa/bcl-gallery/data/data";

const template = "@oe-bcl/bcl-gallery/gallery.html.twig";
const render = (params) => renderTwigFileAsNode(template, params);

describe("OE - Gallery", () => {
  test("default renders correctly", () => {
    expect.assertions(1);
    return expect(render(dataDefault)).resolves.toMatchSnapshot();
  });

  test("renders correctly with title and link", () => {
    expect.assertions(1);
    return expect(
      render({ ...dataDefault, counter: 4 })
    ).resolves.toMatchSnapshot();
  });

  test("renders correctly with title and link", () => {
    expect.assertions(1);
    return expect(
      render({
        ...dataDefault,
        title: "File test title",
        title_tag: "h6",
        title_link: {
          path: "/example.html",
          label: "Link label",
        },
      })
    ).resolves.toMatchSnapshot();
  });
});
