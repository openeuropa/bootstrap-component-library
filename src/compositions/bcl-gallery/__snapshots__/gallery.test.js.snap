// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OE - Gallery default renders correctly 1`] = `
<jest>
  <div
    class="bcl-gallery"
  >
    <div
      class="d-sm-flex flex-sm-row align-items-sm-center"
    >
      <div
        class="flex-sm-grow-1"
      >
        <h2
          class="fw-bold mb-4 bcl-heading"
        >
          Media gallery title
        </h2>
      </div>
      <div
        class="mb-3 mb-sm-0 fw-bold"
      >
        (10) medias
      </div>
    </div>
    <div
      class="bcl-gallery__thumbnails"
    >
      <ul
        class="bcl-gallery__grid"
      >
        <li>
          <a
            class="bcl-gallery__item d-block w-100 img-fluid"
            data-bs-slide-to="0"
            data-bs-target="#carouselGalleryExample"
            href="#"
          >
            <div
              class="bcl-gallery__item-overlay"
              data-bs-target="#modalGalleryExample"
              data-bs-toggle="modal"
            >
              <div
                class="bcl-gallery__item-caption"
              >
                <h5>
                  First slide label
                </h5>
                <div
                  class="bcl-gallery__item-description"
                >
                  <p>
                    Some representative placeholder content for the first slide.
                  </p>
                </div>
              </div>
            </div>
            <img
              alt="First slide"
              src="https://picsum.photos/id/1005/400/400"
            />
          </a>
        </li>
        <li>
          <a
            class="bcl-gallery__item d-block w-100 img-fluid"
            data-bs-slide-to="1"
            data-bs-target="#carouselGalleryExample"
            href="#"
          >
            <div
              class="bcl-gallery__item-overlay"
              data-bs-target="#modalGalleryExample"
              data-bs-toggle="modal"
            >
              <div
                class="bcl-gallery__item-caption"
              >
                <h5>
                  Slide label
                </h5>
                <div
                  class="bcl-gallery__item-description"
                >
                  <p>
                    Some representative placeholder content for the slide.
                  </p>
                </div>
              </div>
            </div>
            <img
              alt="slide"
              src="https://picsum.photos/id/101/400/400"
            />
          </a>
        </li>
        <li>
          <a
            class="bcl-gallery__item d-block w-100 img-fluid"
            data-bs-slide-to="2"
            data-bs-target="#carouselGalleryExample"
            href="#"
          >
            <div
              class="bcl-gallery__item-overlay"
              data-bs-target="#modalGalleryExample"
              data-bs-toggle="modal"
            >
              <span
                class="bcl-gallery__item-play-icon"
              >
                <svg
                  class="bi icon--s"
                >
                  <use
                    xlink:href="/icons.svg#play-fill"
                  />
                </svg>
              </span>
              <div
                class="bcl-gallery__item-caption"
              >
                <h5>
                  Slide label
                </h5>
                <div
                  class="bcl-gallery__item-description"
                >
                  <p>
                    Some representative placeholder content for the slide.
                  </p>
                </div>
              </div>
            </div>
            <img
              alt="slide"
              src="https://picsum.photos/id/1012/400/400"
            />
          </a>
        </li>
        <li>
          <a
            class="bcl-gallery__item d-block w-100 img-fluid"
            data-bs-slide-to="3"
            data-bs-target="#carouselGalleryExample"
            href="#"
          >
            <div
              class="bcl-gallery__item-overlay"
              data-bs-target="#modalGalleryExample"
              data-bs-toggle="modal"
            >
              <div
                class="bcl-gallery__item-caption"
              >
                <h5>
                  Slide label
                </h5>
                <div
                  class="bcl-gallery__item-description"
                >
                  <p>
                    Some representative placeholder content for the slide.
                  </p>
                </div>
              </div>
            </div>
            <img
              alt="Slide"
              src="https://picsum.photos/id/1015/400/400"
            />
          </a>
        </li>
        <li>
          <a
            class="bcl-gallery__item d-block w-100 img-fluid"
            data-bs-slide-to="4"
            data-bs-target="#carouselGalleryExample"
            href="#"
          >
            <div
              class="bcl-gallery__item-overlay"
              data-bs-target="#modalGalleryExample"
              data-bs-toggle="modal"
            >
              <span
                class="bcl-gallery__item-play-icon"
              >
                <svg
                  class="bi icon--s"
                >
                  <use
                    xlink:href="/icons.svg#play-fill"
                  />
                </svg>
              </span>
            </div>
            <img
              alt="slide"
              src="https://picsum.photos/id/1081/400/400"
            />
          </a>
        </li>
      </ul>
      <div
        class="bcl-gallery__thumbnails-collaspe collapse"
        id="collapseGalleryExample"
      >
        <ul
          class="bcl-gallery__grid"
        >
          <li>
            <a
              class="bcl-gallery__item d-block w-100 img-fluid"
              data-bs-slide-to="5"
              data-bs-target="#carouselGalleryExample"
              href="#"
            >
              <div
                class="bcl-gallery__item-overlay"
                data-bs-target="#modalGalleryExample"
                data-bs-toggle="modal"
              >
                <div
                  class="bcl-gallery__item-caption"
                >
                  <h5>
                    Slide label
                  </h5>
                  <div
                    class="bcl-gallery__item-description"
                  >
                    <p>
                      Some representative placeholder content for the slide.
                    </p>
                  </div>
                </div>
              </div>
              <img
                alt="slide"
                src="https://picsum.photos/id/1018/400/400"
              />
            </a>
          </li>
          <li>
            <a
              class="bcl-gallery__item d-block w-100 img-fluid"
              data-bs-slide-to="6"
              data-bs-target="#carouselGalleryExample"
              href="#"
            >
              <div
                class="bcl-gallery__item-overlay"
                data-bs-target="#modalGalleryExample"
                data-bs-toggle="modal"
              >
                <div
                  class="bcl-gallery__item-caption"
                >
                  <h5>
                    Slide label
                  </h5>
                  <div
                    class="bcl-gallery__item-description"
                  >
                    <p>
                      Some representative placeholder content for the slide.
                    </p>
                  </div>
                </div>
              </div>
              <img
                alt="slide"
                src="https://picsum.photos/id/1026/400/400"
              />
            </a>
          </li>
          <li>
            <a
              class="bcl-gallery__item d-block w-100 img-fluid"
              data-bs-slide-to="7"
              data-bs-target="#carouselGalleryExample"
              href="#"
            >
              <div
                class="bcl-gallery__item-overlay"
                data-bs-target="#modalGalleryExample"
                data-bs-toggle="modal"
              >
                <div
                  class="bcl-gallery__item-caption"
                >
                  <h5>
                    Slide label
                  </h5>
                  <div
                    class="bcl-gallery__item-description"
                  >
                    <p>
                      Some representative placeholder content for the slide.
                    </p>
                  </div>
                </div>
              </div>
              <img
                alt="slide"
                src="https://picsum.photos/id/1043/400/400"
              />
            </a>
          </li>
          <li>
            <a
              class="bcl-gallery__item d-block w-100 img-fluid"
              data-bs-slide-to="8"
              data-bs-target="#carouselGalleryExample"
              href="#"
            >
              <div
                class="bcl-gallery__item-overlay"
                data-bs-target="#modalGalleryExample"
                data-bs-toggle="modal"
              >
                <div
                  class="bcl-gallery__item-caption"
                >
                  <h5>
                    Slide label
                  </h5>
                  <div
                    class="bcl-gallery__item-description"
                  >
                    <p>
                      Some representative placeholder content for the slide.
                    </p>
                  </div>
                </div>
              </div>
              <img
                alt="slide"
                src="https://picsum.photos/id/1050/400/400"
              />
            </a>
          </li>
          <li>
            <a
              class="bcl-gallery__item d-block w-100 img-fluid"
              data-bs-slide-to="9"
              data-bs-target="#carouselGalleryExample"
              href="#"
            >
              <div
                class="bcl-gallery__item-overlay"
                data-bs-target="#modalGalleryExample"
                data-bs-toggle="modal"
              >
                <div
                  class="bcl-gallery__item-caption"
                >
                  <h5>
                    Last slide label
                  </h5>
                  <div
                    class="bcl-gallery__item-description"
                  >
                    <p>
                      Some representative placeholder content for the last slide.
                    </p>
                  </div>
                </div>
              </div>
              <img
                alt="slide"
                src="https://picsum.photos/id/1071/400/400"
              />
            </a>
          </li>
        </ul>
      </div>
    </div>
    <div
      class="bg-lighter py-3 px-4 mt-2 rounded text-center"
    >
      <a
        aria-controls="collapseGalleryExample"
        aria-expanded="false"
        class="bcl-gallery__collapse standalone d-none d-sm-inline-block"
        data-bs-toggle="collapse"
        href="#collapseGalleryExample"
        role="button"
      >
        <span
          class="label-collapsed"
        >
          view the full gallery (10)
        </span>
        <span
          class="label-expanded"
        >
          view less
        </span>
        <svg
          class="ms-2-5 bi icon--2xs"
        >
          <use
            xlink:href="/icons.svg#caret-down-fill"
          />
        </svg>
      </a>
      <a
        class="bcl-gallery__mobile-view-more standalone d-inline-block d-sm-none"
        data-bs-slide-to="5"
        data-bs-target="#carouselGalleryExample"
        href="#"
        role="button"
      >
        <span
          data-bs-target="#modalGalleryExample"
          data-bs-toggle="modal"
        >
          <span
            class="label-collapsed"
          >
            view the full gallery (10)
          </span>
          <svg
            class="ms-2-5 bi icon--2xs"
          >
            <use
              xlink:href="/icons.svg#caret-down-fill"
            />
          </svg>
          <span />
        </span>
      </a>
    </div>
    <div
      aria-hidden="true"
      aria-labelledby="modalGalleryExample"
      class="modal"
      data-bs-backdrop="false"
      id="modalGalleryExample"
      tabindex="-1"
    >
      <div
        class="modal-dialog modal-dialog-centered modal-fullscreen"
      >
        <div
          class="modal-content"
        >
          <div
            class="modal-header border-0 justify-content-between"
          >
            <div
              class="modal-title"
            >
              <h3
                class="d-none d-sm-block"
              >
                Media gallery title
              </h3>
            </div>
            <div
              class="carousel-pager"
            >
              <span>
                1
              </span>
               / 10
            
            </div>
            <div
              class="modal-close"
            >
              <button
                aria-label="Close"
                class="btn-close"
                data-bs-dismiss="modal"
                type="button"
              />
            </div>
          </div>
          <div
            class="modal-body d-flex align-items-center justify-content-center"
          >
            <div
              class="carousel slide"
              data-bs-interval="false"
              data-bs-ride="carousel"
              id="carouselGalleryExample"
            >
              <div
                class="carousel-inner"
              >
                <div
                  class="carousel-item active"
                >
                  <img
                    alt="First slide"
                    data-src="https://picsum.photos/id/1005/800/400"
                  />
                  <div
                    class="carousel-caption d-none d-md-block"
                  >
                    <h5>
                      First slide label
                    </h5>
                    <p>
                      Some representative placeholder content for the first slide.
                    </p>
                  </div>
                </div>
                <div
                  class="carousel-item"
                >
                  <img
                    alt="slide"
                    data-src="https://picsum.photos/id/101/800/400"
                  />
                  <div
                    class="carousel-caption d-none d-md-block"
                  >
                    <h5>
                      Slide label
                    </h5>
                    <p>
                      Some representative placeholder content for the slide.
                    </p>
                  </div>
                </div>
                <div
                  class="carousel-item"
                >
                  <iframe
                    data-src="https://www.youtube.com/embed/fgi-GSCB6ho"
                    frameborder="0"
                  />
                </div>
                <div
                  class="carousel-item"
                >
                  <img
                    alt="slide"
                    data-src="https://picsum.photos/id/1015/800/400"
                  />
                  <div
                    class="carousel-caption d-none d-md-block"
                  >
                    <h5>
                      Slide label
                    </h5>
                    <p>
                      Some representative placeholder content for the slide.
                    </p>
                  </div>
                </div>
                <div
                  class="carousel-item"
                >
                  <video
                    controls="controls"
                    poster="https://picsum.photos/id/1081/800/400"
                  >
                    <source
                      src="https://inno-ecl.s3.amazonaws.com/media/videos/big_buck_bunny.mp4"
                      type="video/mp4"
                    />
                    <source
                      src="https://inno-ecl.s3.amazonaws.com/media/videos/big_buck_bunny.webm"
                      type="video/webm"
                    />
                    <track
                      kind="captions"
                      label="English"
                      src="/captions/bunny-en.vtt"
                      srclang="en"
                    />
                    <track
                      kind="captions"
                      label="français"
                      src="/captions/bunny-fr.vtt"
                      srclang="fr"
                    />
                  </video>
                </div>
                <div
                  class="carousel-item"
                >
                  <img
                    alt="Slide"
                    data-src="https://picsum.photos/id/1018/800/400"
                  />
                  <div
                    class="carousel-caption d-none d-md-block"
                  >
                    <h5>
                      Slide label
                    </h5>
                    <p>
                      Some representative placeholder content for the slide.
                    </p>
                  </div>
                </div>
                <div
                  class="carousel-item"
                >
                  <img
                    alt="Slide"
                    data-src="https://picsum.photos/id/1026/800/400"
                  />
                  <div
                    class="carousel-caption d-none d-md-block"
                  >
                    <h5>
                      Slide label
                    </h5>
                    <p>
                      Some representative placeholder content for the slide.
                    </p>
                  </div>
                </div>
                <div
                  class="carousel-item"
                >
                  <img
                    alt="Slide"
                    data-src="https://picsum.photos/id/1043/800/400"
                  />
                  <div
                    class="carousel-caption d-none d-md-block"
                  >
                    <h5>
                      Slide label
                    </h5>
                    <p>
                      Some representative placeholder content for the slide.
                    </p>
                  </div>
                </div>
                <div
                  class="carousel-item"
                >
                  <img
                    alt="First slide"
                    data-src="https://picsum.photos/id/1050/800/400"
                  />
                  <div
                    class="carousel-caption d-none d-md-block"
                  >
                    <h5>
                      Slide label
                    </h5>
                    <p>
                      Some representative placeholder content for the slide.
                    </p>
                  </div>
                </div>
                <div
                  class="carousel-item"
                >
                  <img
                    alt="First slide"
                    data-src="https://picsum.photos/id/1071/800/400"
                  />
                  <div
                    class="carousel-caption d-none d-md-block"
                  >
                    <h5>
                      Last slide label
                    </h5>
                    <p>
                      Some representative placeholder content for the last slide.
                    </p>
                  </div>
                </div>
              </div>
              <button
                class="carousel-control-prev"
                data-bs-slide="prev"
                data-bs-target="#carouselGalleryExample"
                type="button"
              >
                <span
                  aria-hidden="true"
                  class="carousel-control-prev-icon"
                />
                <span
                  class="visually-hidden"
                >
                  Previous
                </span>
              </button>
              <button
                class="carousel-control-next"
                data-bs-slide="next"
                data-bs-target="#carouselGalleryExample"
                type="button"
              >
                <span
                  aria-hidden="true"
                  class="carousel-control-next-icon"
                />
                <span
                  class="visually-hidden"
                >
                  Next
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</jest>
`;
